File: _01_Languages/Java/src/_05_excepciones/_06_ExcepcionPersonalizada.java
package _01_Languages.Java.src._05_excepciones;

/*
 * 🧩 Enunciado:
 * Crea una excepción personalizada llamada EdadInvalidaException.
 * Lanza la excepción si la edad es menor que 0 o mayor que 120.
 */

class EdadInvalidaException extends Exception {
    public EdadInvalidaException(String mensaje) {
        super(mensaje);
    }
}

public class _06_ExcepcionPersonalizada {
    public static void validarEdad(int edad) throws EdadInvalidaException {
        if (edad < 0 || edad > 120)
            throw new EdadInvalidaException("Edad fuera de rango permitido.");
        System.out.println("✅ Edad válida: " + edad);
    }

    public static void main(String[] args) {
        try {
            validarEdad(150);
        } catch (EdadInvalidaException e) {
            System.out.println("❌ Error: " + e.getMessage());
        }
    }
}

File: _01_Languages/CSharp/src/_05_excepciones/_06_ExcepcionPersonalizada.cs
/*
 * 🧩 Enunciado:
 * Crea una excepción personalizada llamada EdadInvalidaException
 * y lánzala si la edad es menor de 0 o mayor de 120.
 */

using System;

class EdadInvalidaException : Exception
{
    public EdadInvalidaException(string mensaje) : base(mensaje) { }
}

class _05_ExcepcionPersonalizada
{
    static void VerificarEdad(int edad)
    {
        if (edad < 0 || edad > 120)
            throw new EdadInvalidaException("La edad introducida no es válida.");
        Console.WriteLine($"Edad válida: {edad}");
    }

    static void Main()
    {
        try
        {
            Console.Write("Introduce tu edad: ");
            int edad = Convert.ToInt32(Console.ReadLine());
            VerificarEdad(edad);
        }
        catch (EdadInvalidaException ex)
        {
            Console.WriteLine("❌ Error personalizado: " + ex.Message);
        }
        catch (FormatException)
        {
            Console.WriteLine("❌ Error: Debes introducir un número válido.");
        }
    }
}
